### PR TITLE
Use roll libs for requests, auth & components

### PR DESCRIPTION
**Pull request by teimurjan**
# Description

The PR adds usage of the Roll NPM packages for request-making, authorization & UI design.

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -1,56 +1,49 @@
...
 
-// update the oauth configuration to log in with a different client
-const oauthConfig = {
-  clientID: config.clientID,
-  issuerURL: config.issuerURL,
-  redirectURL: config.redirectURL,
-  scopes: config.scopes,
+const apiClientParsers = {
+  error: (response) => {
+    if (typeof response.data === "object") {
```
Comment (marlon-wiprud) [inline src/api/index.js:26]: this seems like something that should be part of the api package. This should be the default behavior, which the consumer can hook into if they want to.

File: src/api/index.js [hunk 2]
```diff
@@ -1,56 +1,49 @@
...
-const getCachedOauthTokens = () => {
-  return localStorage.getItem(OAUTH_KEY);
-};
+const authSdk = new auth.AuthSDK(authSdkConfig, localStorage);
 
-const clearCachedOauthTokens = () => {
-  localStorage.removeItem(OAUTH_KEY);
+const apiClientConfig = {
+  getApiUrl: () => config.apiURL,
+  getAuthorization: () => {
```
Comment (marlon-wiprud) [inline src/api/index.js:17]: the consumer shouldn't be responsible for defining how the auth token is formatted

Comment (marlon-wiprud) [reply src/api/index.js:17]: can we set this up such that the api package expects an authorization interface, which the auth sdk fulfills out of the box? Similar to the way that the auth sdk expects a persistence interface, which local storage fulfills.

File: src/api/index.js [hunk 3]
```diff
@@ -1,56 +1,49 @@
...
+  },
 };
 
-// add custom caching implementation to store tokens and refresh user sessions
-const authCacheConfig = {
-  setCache: cacheOauthTokens,
-  getCache: getCachedOauthTokens,
-  clearCache: clearCachedOauthTokens,
+const apiClientHandlers = {
+  refresh: authSdk.refreshTokens,
```
Comment (marlon-wiprud) [inline src/api/index.js:40]: if we use the auth interface, this should be part of that interface rather than a separate config

**Review by marlon-wiprud — CHANGES_REQUESTED** [review]

File: src/components/sessionManager.js [hunk 4]
```diff
@@ -1,60 +1,66 @@
...
-      const urlParams = new URLSearchParams(window.location.search);
-      if (urlParams.has("code")) {
-        rollAPI.authentication.initializeSession(
-          window.location.search,
-          handleSuccess,
-          handleFail
-        );
+    const initializeNewSession = async () => {
+      try {
+        const urlParams = new URLSearchParams(window.location.search);
```
Comment (marlon-wiprud) [inline src/components/sessionManager.js:27]: perhaps we can provide a function `makeSessionFromUrl` so the consumer doesn't need to parse it themselves?

**Review by marlon-wiprud — CHANGES_REQUESTED** [review]

File: src/api/index.js (hunk 1 above)
Comment (teimurjan) [reply src/api/index.js]: ✅

File: src/components/sessionManager.js (hunk 4 above)
Comment (teimurjan) [reply src/components/sessionManager.js]: It's now hidden inside the session manager package.

**Review by marlon-wiprud — APPROVED** [review]

**Pull request merged**